cmd/stream-commp/main.go

```diff
@@ -176,3 +175,16 @@ func scanInputStream(streamBuf *bufio.Reader) (cnt int64, res string) {
 	}
 	return
 }
+
+// Using io.Discard in the various Copy() invocations above results in invoking
+// https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/io/io.go;l=647-661
+// which in turn is bound by this limit:
+// https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/io/io.go;l=642
+// resulting in micro-writes into the hasher
+// Use a dumb discarder instead
+type unsmartDiscard struct{}
+
+var uDiscard io.Writer = unsmartDiscard{}
```

Comment:
Should we pull this out as a standalone library? I reckon we have this pattern of copying into io.Discard in a few places (carlet probably also has this)

Reply:
Seems too... narrow/specific. Perhaps raising a bugreport against the go stdlib is more appripriate. @AnomalRoil what are your thoughts on this? ( the `io.Discard` weirdness we discussed in slack )

Reply:
@AnomalRoil if this is an option: I think this would be best yes. It's an obscure corner-ish case that'd be difficult for an outsider to convey.